core/app/models/spree/stock_item.rb

```diff
@@ -96,12 +96,19 @@ def process_backorders(number)
       end
 
       def conditional_variant_touch
-        # the variant_id changes from nil when a new stock location is added
-        stock_changed = (count_on_hand_changed? && count_on_hand_change.any? { |c| c < 1 }) || variant_id_changed?
-
-        if !Spree::Config.binary_inventory_cache || stock_changed
-          variant.touch
+        # only warn if store is setting binary_inventory_cache (default = false)
+        if Spree::Config.binary_inventory_cache
+          Spree::Config.inventory_cache_threshold = 1
```

Comment:
This should not change the value of `inventory_cache_threshold`. If I understand this correctly maybe it should read:

```
def conditional_variant_touch
  variant.touch if count_breaks_threshold?
end

def count_breaks_threshold?
  inventory_cache_threshold &&
  ...
end

private
def inventory_cache_threshold
  if Spree::Config.binary_inventory_cache
    ActiveSupport::Deprecation.warn "Spree::Config.binary_inventory_cache=true is DEPRECATED. Instead use Spree::Config.inventory_cache_threshold=1"
    1
  else
    Spree::Config.inventory_cache_threshold
  end
end
```

That way we're using the correct values without re-assigning the preferences.